src/types.ts

```diff
@@ -40,8 +41,8 @@ type IgnoredTypes = Promise<unknown> | ((...args: unknown[]) => unknown);
 export type ChangeObject<Type> = {
   [Property in keyof Type as Exclude<
     Property,
-    Type[Property] extends IgnoredTypes ? Property : never
-  >]: Type[Property] extends Date ? number : Type[Property];
+    Required<Type>[Property] extends IgnoredTypes ? Property : never
+  >]: Required<Type>[Property] extends Date ? number : Type[Property];
```

Comment:
If propery was declared as `date?: Date` in type the `ChangeObject` util would not cast it to `number`. With this change it does so correctly.